design/general_refactorings.md

```diff
@@ -0,0 +1,381 @@
+## CLI Refactor
+
+
+The purpose of this document is to outline the parts of the Python CLI/SDK code that we would like to refactor as well as provide a design for each point. Through this refactor, we hope that the code will be more readable, and easier to contribute to and maintain.
+
+
+
+### Improve Naming
+
+------
+
+##### Proposal
+
+Review the names of files, functions.. and rename them to more readable names.
+
+##### Current State
+
+Not all the names are clear now. For example with `class Resource` it is hard to understand which resource we mean.
+
+##### Recommendation
+
+Go over all naming in the code and rename when need. If we rename  `class Resource` to `class ConjurResource` developer who is reading the code will understand we are referencing a specific resource, a Conjur resource. The same applies for the names of files in the repository.
+
+### Organize directory structure
+
+------
+
+##### Proposal
+
+Update the new current directory structure so that it is divided by the types of files and their responsibilities. For example, all `Controller` classes will be placed in a single directory.
+
+##### Current State
+
+Currently, the directory structure resembles the following
+
+| Directory Name | Purpose                           |
+| :------------- | :-------------------------------- |
+| host           | files related to host command     |
+| init           | files related to init command     |
+| list           | files related to list command     |
+| login          | files related to login command    |
+| logout         | files related to logout command   |
+| policy         | files related to policy command   |
+| user           | files related to user command     |
+| variable       | files related to variable command |
+
+The following files are without parent directory : api.py, client.py, argparse_wrapper.py, cli.py , client.py, config.py, constants.py, credentials_data.py, credentials_form_file.py, endpoint.py, errors.py, http_wrapper.py, resource.py, ssl_service.py, version.py
+
+##### Recommendation
+
+I recommend that we organize the files by their types and not by the commands they represent.
+
+| Directory Name | Purpose                                                      |
+| :------------- | :----------------------------------------------------------- |
+| wrappers       | wrappers and utils of diffrent python libs :  arg_parse_wrapper, http_wrapper |
+| conjur_api     | files that talk directly with conjur. For Example   `conjur_client.py (conjur.py) , conjur_api.py (api.py), ssl_client.py (ssl_service)` |
```

Comment:
files that communicate directly with Conjur. For example, `conjur_client.py (conjur.py) , conjur_api.py (api.py), ssl_client.py (ssl_service)`

I feel like I've commented on these a few times already 🤔

Reply:
Sorry if im missed. Fixed it now.